image/Makefile

```diff
@@ -18,34 +29,30 @@ help: ## show make targets
 .PHONY: clean
 clean: ## remove build artifacts
 	-$(RM) $(ENGINE_DIR)/Dockerfile.engine
-	-docker rmi $(DOCKER_HUB_ORG)/$(ENGINE_IMAGE):$(STATIC_VERSION)
-	-rm -f image-linux
+	-$(RM) $(ENGINE_DIR)/Dockerfile.engine-dm
+	-docker rmi $(IMAGE_WITH_TAG)
+	-rm -f image-linux-*
 
 .PHONY: image
 image: image-linux
 
-
-$(ENGINE_DIR)/Dockerfile.engine:
-	cp Dockerfile.engine $(ENGINE_DIR)
+$(ENGINE_DIR)/Dockerfile.%: Dockerfile.%
+	cp $< $@
 
 # builds across multiple archs because the base images
 # utilize manifests
 image-linux: $(ENGINE_DIR)/Dockerfile.engine
-	docker build -t $(DOCKER_HUB_ORG)/$(ENGINE_IMAGE):$(STATIC_VERSION).$(ARCH) \
-		--build-arg GO_IMAGE="$(GO_IMAGE)" \
-		--build-arg VERSION="$(STATIC_VERSION)" \
-		--build-arg GITCOMMIT="$$(cd $(ENGINE_DIR) && git rev-parse --short=7 HEAD)" \
-		--build-arg BUILDTIME="$(BUILDTIME)" \
-		--build-arg PLATFORM="$(PLATFORM)" \
-		--build-arg PRODUCT="$(PRODUCT)" \
-		--build-arg DEFAULT_PRODUCT_LICENSE="$(DEFAULT_PRODUCT_LICENSE)" \
-		--file $< $(ENGINE_DIR)
-	echo $(DOCKER_HUB_ORG)/$(ENGINE_IMAGE):$(STATIC_VERSION).$(ARCH) > $@
+	./image_exists $(IMAGE_WITH_TAG) || $(IMAGE_BUILD)
+	echo $(IMAGE_WITH_TAG) > $@
+
+image-linux-dm: $(ENGINE_DIR)/Dockerfile.engine-dm
+	$(warning WARN: This image will be named: $(IMAGE_WITH_TAG)-dm)
+	./image_exists $(IMAGE_WITH_TAG)-dm || $(IMAGE_BUILD)
+	echo $(IMAGE_WITH_TAG) > $@
 
 engine-$(ARCH).tar: image-linux
 	docker save -o $@ $$(cat $<)
 
-
 .PHONY: release
 release:
-	docker push $(DOCKER_HUB_ORG)/$(ENGINE_IMAGE):$(STATIC_VERSION).$(ARCH)
+	docker push $(IMAGE_WITH_TAG)
```

Comment:
Does the release target need to account for whether the image name has a `-dm` appended to the end of it?